mlir/lib/Dialect/Presburger/PresburgerOps.cpp

```diff
@@ -49,6 +49,31 @@ static LogicalResult verify(SetOp op) {
   return success();
 }
 
+// Presburger pw expr
+
+static void print(OpAsmPrinter &printer, PwExprOp op) {
+  printer << "presburger.pwExpr " << op.pwExprAttr();
+}
+
+static ParseResult parsePwExprOp(OpAsmParser &parser, OperationState &result) {
+  PresburgerPwExprAttr pwExpr;
+
+  if (parser.parseAttribute(pwExpr, "pwExpr", result.attributes))
+    return failure();
+
+  // TODO Currently we inherit the type from the PresburgerPwExprAttr, I'm
+  // not sure if this is desirable.
+  Type outType = pwExpr.getType();
+
+  parser.addTypeToList(outType, result.types);
+  return success();
+}
+
+static LogicalResult verify(PwExprOp op) {
+  // TODO do we need something here?
```

Comment:
We should theoretically check if the domains are disjunct somewhere. I'm just not sure if this is too expensive

Reply:
We could keep it for now without verification and discuss this in the next meeting. Or do you, @tobiasgrosser , have a strong opinion on this one. 
Note that we also need to specify what happens if the domains do not cover the complete space, i.e. what the expression should evaluate to in such cases.

Reply:
I don't have a strong opinion. ;-)